src/rgw/rgw_op.cc

```diff
@@ -5035,13 +5034,15 @@ void RGWPutACLs::execute()
     if (op_ret < 0)
       return;
 
-    new_data = strdup(ss.str().c_str());
-    free(data);
-    data = new_data;
     len = ss.str().size();
+    if (data) {
+      data.release();
+    }
+    data = std::make_unique<char[]>(len + 1);
+    strcpy(data.get(), ss.str().c_str());
```

Comment:
@cbodley : This is the cleanest way I could think of,  please let me know if you know of something better.

Reply:
I think the only issue was mixing allocators, so I would imagine this resolves it